Create admin and customer action:

quynhtrangluong opened the pull request:
+ Create print all information, print selected information and delete selected information of customer( with admin)
+ Create print customer's information, delete customer's information (with customer)

quynhtrangluong committed:
Create admin and customer action:
+ Create print all information, print selected information and delete selected information of customer( with admin)
+ Create print customer's information, delete customer's information (with customer)

daothanhbinh20 commented on AccountService/src/CustomerService.java:
```diff
@@ -24,5 +24,15 @@ public void createData() {
 	}
-}
+	public void printAllCustomersInfo() {
+		for (int i=0; i<userList.size(); i++) {
```
Change this into For each  loop instead - for (User user : userLIst) {//CODE}

daothanhbinh20 commented on AccountService/src/CustomerService.java:
```diff
@@ -24,5 +24,15 @@ public void createData() {
 	public void addUser(User user) {
```
Use final for parameters here

daothanhbinh20 commented on AccountService/src/LoginService.java:
```diff
@@ -17,4 +17,38 @@ public boolean verifyUser(final String username,final String password) {
+		return false;
+	}
+
+	public void printSelectCustomerInfo(String username) {
```
`final` for parameter username

daothanhbinh20 commented on AccountService/src/LoginService.java:
```diff
@@ -17,4 +17,38 @@ public boolean verifyUser(final String username,final String password) {
+		}
+	}
+
+	public void deleteSelectCustomerInfo(String username) {
```
Final for params here as well

daothanhbinh20 commented on AccountService/src/UserInputService.java:
```diff
@@ -68,10 +76,57 @@ public void mainUserInputFlow() {
+			System.out.println("2. Delete customer's information");
+			System.out.println("3. Print selected customer details");
+			System.out.println("Please choose a number for the action you want");
+			String adminSelectOption;
+			adminSelectOption = userInput.nextLine();
```
Does not need two line - String adminSelectOption = userInput.nextLine();

daothanhbinh20 commented on AccountService/src/UserInputService.java:
```diff
@@ -68,10 +76,57 @@ public void mainUserInputFlow() {
+				System.out.println("Please choose customer you want to delete");
+				System.out.println("Customer name: ");
+				String deleteCustomer;
+				deleteCustomer = userInput.nextLine();
```
Can be one line innitialization as above

daothanhbinh20 commented on AccountService/src/UserInputService.java:
```diff
@@ -68,10 +76,57 @@ public void mainUserInputFlow() {
+			System.out.println("Please choose a number for the action you want");
+			String adminSelectOption;
+			adminSelectOption = userInput.nextLine();
+			if(adminSelectOption.equals("1")){
+				customerService.printAllCustomersInfo();
+			}
+			if(adminSelectOption.equals("2")){
+				System.out.println("Please choose customer you want to delete");
+				System.out.println("Customer name: ");
+				String deleteCustomer;
+				deleteCustomer = userInput.nextLine();
+				if(loginService.verifyUsername(deleteCustomer)){
+					loginService.deleteSelectCustomerInfo(deleteCustomer);
+				}
+				else {
+					System.out.println("Account name does not exist");
+				}
+			}
+			if(adminSelectOption.equals("3")){
+				System.out.println("PLease choose customer you want to print out");
+				String printCustomer;
+				printCustomer = userInput.nextLine();
+				if(loginService.verifyUsername(printCustomer)){
+						loginService.printSelectCustomerInfo(printCustomer);
+				}
+			}
```
Use switch statement for `adminSelectOption`. This avoids repetition of adminSelectOption.equals()

daothanhbinh20 commented on AccountService/src/UserInputService.java:
```diff
@@ -68,10 +76,57 @@ public void mainUserInputFlow() {
+			System.out.println("Please choose a number for the action you want");
+			String customerSelectOption;
+			customerSelectOption = userInput.nextLine();
+			if(customerSelectOption.equals("1")){
+				loginService.printSelectCustomerInfo(currentUser.getUserName());
+			}
+			if(customerSelectOption.equals("2")){
+				loginService.deleteSelectCustomerInfo(currentUser.getUserName());
+			}
```
Use switch statement for adminSelectOption. This avoids repetition of adminSelectOption.equals()

daothanhbinh20 commented on AccountService/src/UserInputService.java:
```diff
@@ -68,10 +76,57 @@ public void mainUserInputFlow() {
+			System.out.println("2. Delete customer's information");
+			System.out.println("3. Print selected customer details");
+			System.out.println("Please choose a number for the action you want");
+			String adminSelectOption;
+			adminSelectOption = userInput.nextLine();
+			if(adminSelectOption.equals("1")){
+				customerService.printAllCustomersInfo();
+			}
+			if(adminSelectOption.equals("2")){
+				System.out.println("Please choose customer you want to delete");
+				System.out.println("Customer name: ");
+				String deleteCustomer;
+				deleteCustomer = userInput.nextLine();
+				if(loginService.verifyUsername(deleteCustomer)){
+					loginService.deleteSelectCustomerInfo(deleteCustomer);
+				}
+				else {
+					System.out.println("Account name does not exist");
+				}
+			}
+			if(adminSelectOption.equals("3")){
+				System.out.println("PLease choose customer you want to print out");
+				String printCustomer;
+				printCustomer = userInput.nextLine();
+				if(loginService.verifyUsername(printCustomer)){
+						loginService.printSelectCustomerInfo(printCustomer);
+				}
+			}
 		}
+		else{
+			System.out.println("1. Print your information");
+			System.out.println("2. Delete your information");
+			System.out.println("Please choose a number for the action you want");
+			String customerSelectOption;
+			customerSelectOption = userInput.nextLine();
+			if(customerSelectOption.equals("1")){
+				loginService.printSelectCustomerInfo(currentUser.getUserName());
+			}
+			if(customerSelectOption.equals("2")){
+				loginService.deleteSelectCustomerInfo(currentUser.getUserName());
```
I think we should put this entire block of code into another function, so we don't end up with a massive function. maybe be naming it customerActionInput()

quynhtrangluong committed:
Create test for LoginService Class:
+ Test for VerifyUser
+ Test for VerifyUsername
Improve the code after code review:
+ Change from for loop to for each loop
+ Add final to some parameters
+ Simply some code
+ Change to Switch statement for adminSelectOption and customerSelectOption
+ Bring the adminSelectOption and customerSelectOption to a new code block is customerActionInput for reduce massive function.

quynhtrangluong committed:
Create test for LoginService Class:
+ Test for VerifyUser
+ Test for VerifyUsername
Improve the code after code review:
+ Change from for loop to for each loop
+ Add final to some parameters
+ Simply some code
+ Change to Switch statement for adminSelectOption and customerSelectOption
+ Bring the adminSelectOption and customerSelectOption to a new code block is customerActionInput for reduce massive function.
